json-config.hpp

```diff
@@ -156,6 +160,47 @@ class JsonConfig
         }
         confFile.clear();
 
+        try
+        {
+            auto configMapJsonPath = fs::path{confBasePath} / configMapJson;
+
+            // based on device tree check if config file exists for the model.
+            if (fs::exists(modelFilePath) && fs::exists(configMapJsonPath))
+            {
+                std::ifstream modelFile(modelFilePath);
+                std::ifstream configMapFile(configMapJsonPath);
+                if (modelFile.is_open() && configMapFile.is_open())
+                {
+                    std::string model;
+                    std::getline(modelFile, model, '\0');
+
+                    auto configMap = nlohmann::json::parse(configMapFile);
+
+                    if (configMap.contains("IBMCompatibleSystems"))
+                    {
+                        if (configMap["IBMCompatibleSystems"].contains(model))
+                        {
+                            std::vector<std::string> name;
+                            name.push_back(
+                                configMap["IBMCompatibleSystems"][model]);
+                            if (filePathExists(name))
+                            {
+                                // Use the first config file found at a listed
+                                // location
```

Comment:
I think you need to set confFile here, like it does up above on 156, and then it can delete the EM match on line 55.

Reply:
The setting of conf file at line 156 is static. Whereas in this case, here it is based on value read from model. Using which the function "filePathExists" can set the conf file dynamically.

Reply:
ok, yea  I see it now, filePathExists also sets it.